src/arch/linux/mapping/mapping.c

```diff
@@ -747,6 +747,8 @@ static int do_map_hwram(struct hardware_ram *hw)
   int cap = MAPPING_KMEM;
   if (hw->default_vbase != (dosaddr_t)-1)
     cap |= MAPPING_LOWMEM;
+  else if (!config.dpmi)
```

Comment:
Oh, is this only to shrink the mem_pool?
It doesn't look right.
If we have a hwram somewhere, its still
accessing via int15, regardless of DPMI.
You can't just skip it.

Reply:
for int15 hwram physaddr_to_unixaddr resolves straight to SHM memory, not main_pool.

This works fine, even with my follow up patch with $_dpmi=(0) where main_pool is only 0x110000, with himem.sys loaded, and XMS is provided via int15.